File: 프로그래머스/lv2/12909. 올바른 괄호/올바른 괄호.js
function solution(s) {
  let st = 0;
  for (let char of s) {
    if (char === "(") {
      st++;
    } else if (char === ")" && st) {
      st--;
    } else {
      return false;
    }
  }
  return st ? false : true;
}

File: 프로그래머스/lv2/12909. 올바른 괄호/올바른 괄호.py
def solution(s):
    ans = 0
    # ")"로 시작하는 문자열을 사전 차단
    if s[0] == ")":
        return False
    
    for c in s:
        if c == "(":
            ans += 1
        elif c == ")":
            ans -= 1 
        if ans < 0 :
            return False    
    return True if ans == 0 else False